fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergPartitionInfo.java

```diff
@@ -0,0 +1,71 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.datasource.iceberg;
+
+import org.apache.doris.catalog.PartitionItem;
+
+import com.google.common.collect.Maps;
+
+import java.util.Map;
+import java.util.Set;
+
+public class IcebergPartitionInfo {
+    private final Map<String, PartitionItem> nameToPartitionItem;
+    private final Map<String, IcebergPartition> nameToIcebergPartition;
+    private final Map<String, Set<String>> nameToIcebergPartitionNames;
+
+    public IcebergPartitionInfo() {
+        this.nameToPartitionItem = Maps.newHashMap();
+        this.nameToIcebergPartition = Maps.newHashMap();
+        this.nameToIcebergPartitionNames = Maps.newHashMap();
+    }
+
+    public IcebergPartitionInfo(Map<String, PartitionItem> nameToPartitionItem,
+                               Map<String, IcebergPartition> nameToIcebergPartition,
+                                Map<String, Set<String>> nameToIcebergPartitionNames) {
+        this.nameToPartitionItem = nameToPartitionItem;
+        this.nameToIcebergPartition = nameToIcebergPartition;
+        this.nameToIcebergPartitionNames = nameToIcebergPartitionNames;
+    }
+
+    public Map<String, PartitionItem> getNameToPartitionItem() {
+        return nameToPartitionItem;
+    }
+
+    public Map<String, IcebergPartition> getNameToIcebergPartition() {
+        return nameToIcebergPartition;
+    }
+
+    public long getLatestSnapshotId(String partitionName) {
+        Set<String> icebergPartitionNames = nameToIcebergPartitionNames.get(partitionName);
+        if (icebergPartitionNames == null) {
```

Comment:
What situation would be null

Reply:
Here is an example:
Iceberg table has 3 partitions, their name to ranges are like:
p1 -> [0, 10)
p2 -> [10, 30)
p3 -> [20, 30)

In Doris, p2 and p3 should be merged to one partition p2 -> [10, 30)。So doris has the 2 partitions:
p1 -> [0, 10)
p2 -> [10, 30)

In this case, we need a map nameToIcebergPartitionNames to record that p2 is actually mapped to p2 and p3 in Iceberg:
nameToIcebergPartitionNames : p2 -> {p2, p3}
The partition p1 in doris is mapped to p1 in Iceberg (1 to 1 map), so we don't keep any mapping in nameToIcebergPartitionNames for p1. So when we try to get snapshot id for p1, nameToIcebergPartitionNames.get("p1") will return null.